catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/ResourceCache.java

```diff
@@ -308,26 +310,25 @@ public Resource getValid(String key, Metacard latestMetacard) {
         // cache directory has had files deleted from it.
         if (cachedResource != null) {
             if (!validateCacheEntry(cachedResource, latestMetacard)) {
-                throw new IllegalArgumentException(
-                        "Entry found in cache was out-of-date or otherwise invalid.  Will need to be re-cached.  Entry key: "
+                LOGGER.info(
+                        "Entry found in cache was out-of-date or otherwise invalid.  Will need to be re-cached.  Entry key: {} "
                                 + key);
+                return null;
             }
 
             if (cachedResource.hasProduct()) {
-                LOGGER.debug("EXITING: get() for key {}", key);
+                LOGGER.trace("EXITING: get() for key {}", key);
                 return cachedResource;
             } else {
-                LOGGER.debug(
-                        "Entry found in the cache, but no product found in cache directory for key = {}",
-                        key);
                 cache.remove(key);
-                throw new IllegalArgumentException(
-                        "Entry found in the cache, but no product found in cache directory for key = "
+                LOGGER.info(
+                        "Entry found in the cache, but no product found in cache directory for key = {} "
                                 + key);
+                return null;
             }
         } else {
-            LOGGER.debug("No product found in cache for key = {}", key);
-            throw new IllegalArgumentException("No product found in cache for key = " + key);
+            LOGGER.info("No product found in cache for key = {}", key);
+            return null;
```

Comment:
Why did we change this to return null instead of throwing an exception. I'm not the biggest fan of this pattern. Now everywhere you call getValid, you need to check that it is null as opposed to just knowing that it is non-null.